jax/_src/dtypes.py

```diff
@@ -458,18 +458,22 @@ def isdtype(dtype: DTypeLike, kind: str | DType | tuple[str | DType, ...]) -> bo
     True or False
   """
   the_dtype = np.dtype(dtype)
-  kind_tuple: tuple[DType | str, ...] = kind if isinstance(kind, tuple) else (kind,)
-  options: set[DType] = set()
+  kind_tuple: tuple[DTypeLike, ...] = kind if isinstance(kind, tuple) else (kind,)
+  options: set[DTypeLike] = set()
```

Comment:
`options` should still be `set[DType]`. `DTypeLike` is not hashable in general, so it can't be used in a set.